addons/auth_signup/controllers/main.py

```diff
@@ -41,6 +41,9 @@ def web_auth_signup(self, *args, **kw):
 
         if 'error' not in qcontext and request.httprequest.method == 'POST':
             try:
+                if not request.env['ir.http']._verify_request_recaptcha_token('signup'):
```

Comment:
`_verify_request_recaptcha_token` is defined for the first time in `google_recaptcha`, and `auth_signup` doesn't depend on `google_recaptcha`, it's therefore possible to have `auth_signup` without `google_recaptcha` and hence `_verify_request_recaptcha_token` wouldn't be defined.